aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -211,13 +303,91 @@ impl BaseAnonCreds for IndyCredxAnonCreds {
         sig_type: Option<&str>,
         config_json: &str,
     ) -> VcxCoreResult<(String, String)> {
-        let _ = (issuer_did, schema_json, tag, sig_type, config_json);
-        Err(unimplemented_method_err("credx issuer_create_and_store_credential_def"))
+        let issuer_did = issuer_did.to_owned().into();
+        let schema = serde_json::from_str(schema_json)?;
+        let sig_type = sig_type.map(serde_json::from_str).unwrap_or(Ok(SignatureType::CL))?;
+        let config = serde_json::from_str(config_json)?;
+
+        let schema_seq_no = match &schema {
+            Schema::SchemaV1(s) => s.seq_no,
+        };
+
+        let cred_def_id =
+            credx::issuer::make_credential_definition_id(&issuer_did, schema.id(), schema_seq_no, tag, sig_type)?;
+
+        let options = json!({
+            "retrieve_type": false,
+            "retrieve_value": true,
+            "retrieve_tags": false,
+        })
+        .to_string();
+
+        // If cred def already exists, return it
+        if let Ok(cred_def) = self
+            .wallet
+            .get_wallet_record(CATEGORY_CRED_DEF, &cred_def_id.0, &options)
+            .await
+        {
+            return Ok((cred_def_id.0, cred_def));
+        }
+
+        // Otherwise, create cred def
+        let (cred_def, cred_def_priv, cred_key_correctness_proof) =
```

Comment:
Just observation for other reviewers and myself

The function here has following signature
```rust
pub fn create_credential_definition(
    origin_did: &DidValue,
    schema: &Schema,
    tag: &str,
    signature_type: SignatureType,
    config: CredentialDefinitionConfig,
) -> Result<(
    CredentialDefinition,
    CredentialDefinitionPrivate,
    CredentialKeyCorrectnessProof,
)> {
```

In vdrtools, there is similar function returning similar stuff:
```rust
async fn _create_credential_definition(
        &self,
        attr_names: &AttributeNames,
        support_revocation: bool,
    ) -> IndyResult<(
        CredentialDefinitionData,
        CredentialPrivateKey,
        CredentialKeyCorrectnessProof,
    )> 
```
and the function is used similarly. But you can note that credx one returns `CredentialDefinition` while vdrtools returns `CredentialDefinitionData`. 

This is what credx structure looks like
```rust
#[derive(Debug)]
#[cfg_attr(feature = "serde", derive(Serialize, Deserialize), serde(tag = "ver"))]
pub enum CredentialDefinition {
    #[cfg_attr(feature = "serde", serde(rename = "1.0"))]
    CredentialDefinitionV1(CredentialDefinitionV1),
}

pub struct CredentialDefinitionV1 {
    pub id: CredentialDefinitionId,
    pub schema_id: SchemaId,
    #[cfg_attr(feature = "serde", serde(rename = "type"))]
    pub signature_type: SignatureType,
    pub tag: String,
    pub value: CredentialDefinitionData,
}

#[derive(Debug)]
#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
pub struct CredentialDefinitionData {
    pub primary: ursa_cl!(CredentialPrimaryPublicKey),
    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
    pub revocation: Option<ursa_cl!(CredentialRevocationPublicKey)>,
}

pub const CL_SIGNATURE_TYPE: &str = "CL";

#[derive(Copy, Clone, Debug, PartialEq, Eq)]
#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
pub enum SignatureType {
    CL,
}
```

And this is what `vdrtools`'s `CredentialDefinitionData` looks like
```
#[derive(Debug, Serialize, Deserialize)]
pub struct CredentialDefinitionData {
    pub primary: CredentialPrimaryPublicKey,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub revocation: Option<CredentialRevocationPublicKey>,
}
``` 
So essentially, `credx` returns richer structure with more metadata, while `vdrtools` returned only the core piece - `CredentialDefinitionData` which is the same between the 2 implementations.

Reply:
In order to implement migration of old data to new, we will have to map out the `vdrtools`'s `CredentialDefinitionData` to credx's `CredentialDefinition`. 
Since the additional metadata contained in credx datastructure can be known, this should be pretty fairly doable

Reply:
The trait method now replicates the old behavior.

Reply:
I'm actually going to remove this commit because it looks like this is wrong. The `_create_credential_definition` function returns `CredentialDefinitionData`, but that data is then used to construct an actual `CredentialDefinition` (parent enum), which is what gets returned.